apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Plugin Name: Full Site Editing
+ */
+
+class WPCOM_Full_Site_Editing {
+	static $initialized = false;
+
+	function __construct() {
+		if ( self::$initialized ) {
+			return;
+		}
+		self::$initialized = true;
+
+		$this->register_script_and_style();
```

Comment:
```suggestion
		add_action( 'enqueue_block_editor_assets', array( $this, 'register_script_and_style' ) );
```

Otherwise there is a PHP notice about running `wp_register_script` too early. We'll probably want to have a separate method down the road for non-editor stuff that hooks into `wp_enqueue_scripts` for example, but this works for now.

Reply:
I think `enqueue_block_editor_assets` is supposed to be used for scripts that extend the editor itself, not blocks.

Blocks just need a `wp_register_script` (which doesn't enqueue the script), and define as dependency in the `editor_script` option of `register_block_type`.
When the block type is created, the script gets automatically enqueued (if still needed).

Although, I should change my error reporting and check that notice. 🤔